### DISCUSSION TITLE
customize exported CSV / EXCEL file

### DISCUSSION THREAD

**Discussion by ramillionaire** [kimai/kimai#3121] [Questions & Answers] [answered]
Hi , is it possible to customize the exported CSV /  Excel file? 
Currently it loads everything from the timesheet. We only need some data to be displayed from the timesheet to the exported excel / CSV file.  

Of course i could remove the unnecessary data manually from the excel file  but there would be too many exported file per day so it would be very nice to set this directly.

**Comment by kevinpapst** [comment, marked as answer]
Simpler is to write your own PDF expert template. But it sounds as if PDF is not your thing.

Changing the columns it not possible via UI or config file, only by creating some code:

You would have to write your own ExportRenderer that extends AbstractSpreadsheetRenderer and then [change the columns](https://github.com/kevinpapst/kimai2/blob/master/src/Export/Base/AbstractSpreadsheetRenderer.php#L73).

**Reply by ramillionaire** [reply]
Mein/ Unser Ziel ist es, einige Spalten aus dem Export zu entfernen (Wir benötigen nur 5 Daten/ Spalten die er exportieren soll)

- So wie ich das sehe , könnte ich doch einfach die unerwünschten Spalten aus der AbstractSpreadsheetRenderer auskommentieren/entfernen ohne ein zusätzlichen ExportRenderer zu schreiben.  Der CSV bzw. der  ExcelRenderer existieren bereits.

Sorry, falls ich jetzt hier doch was übersehen habe und es doch nicht möglich ist.

**Reply by kevinpapst** [reply]
Testing = your time, asking = my time. I do prefer the first one.

**Reply by Sarmingsteiner** [reply]
@ramillionaire : We wrote a script in Excel (VBA) for that, unfortunately it got lost :-(
